api/core/audio.js

```diff
@@ -0,0 +1,252 @@
+ *
+ * @constructor
+ * @param {Number} frequency - The mixing frequency in Hz.
+ * @param {Number} bits - The bit depth at which to play sounds.
+ * @param {Number} channels - The number of speaker channels supported by the mixer.
+ */
+function Mixer(frequency, bits, channels)
+{
+	/**
+	 * The mixer volume.
+	 *
+	 * The mixer volume is multiplied with the volume of each individual sounds
+	 * to determine how loud each one is.
+	 */
+	this.volume = 1.0;
+}
+exports.Mixer = Mixer;
+
+/**
+ * Constructs a Sound object, which represents an audio stream backed by a
+ * sound file.  Sounds are fully seekable and the engine automatically keeps
+ * the stream fed using audio data from the underlying file.
+ *
+ * @constructor
+ * @param {String} path - The path to the sound file, e.g. `sounds/munch.wav`.
+ */
+function Sound(path)
+{
+	/**
+	 * The length of the sound in seconds.
+	 *
+	 * @type {Number}
+	 * @readonly
+	 */
+	this.length = 0.0;
+
+	/**
+	 * The Mixer currently being used to the play the sound. `null` if the
+	 * sound is stopped.
+	 *
+	 * @readonly
+	 * @type {Mixer}
+	 */
+	this.mixer = null;
+	
+	/**
+	 * The SphereFS path of the sound.
+	 *
+	 * @readonly
+	 * @type {String}
+	 */
+	this.path = path;
+
+	/**
+	 * Gets or sets the sound's L/R balance.
+	 *
+	 * Ranges from -1.0 to 1.0.
+	 *
+	 * @type {Number}
+	 */
+	this.pan = 0.0;
+
+	/**
+	 * Whether or not the sound is currently playing.
+	 *
+	 * @readonly
+	 * @type {Boolean}
+	 */
+	this.playing = false;
+
+	/**
+	 * The seek position in the sound, in seconds.
+	 *
+	 * @type {Number}
+	 */
+	this.position = 0.0;
+
+	/**
+	 * Gets or sets whether the sound will repeat endlessly.
+	 *
+	 * @type {Boolean}
+	 */
+	this.repeat = false;
+
+	/**
+	 * Gets or sets the volume of the sound.
+	 *
+	 * 1.0 is full volume, 0.0 is silent. This is multiplied by the mixer
+	 * volume to determine the final volume level.
+	 *
+	 * @type {Number}
+	 */
+	this.volume = 1.0;
+
+	/**
+	 * Gets or sets the pitch of the sound.
+	 *
+	 * Must not be negative.
+	 *
+	 * @type {Number}
+	 */
+	this.pitch = 1.0;
+
+	/**
+	 * Start playback of the sound.
+	 *
+	 * If the sound is already playing, it will be stopped and playback will
+	 * begin from 0.0s on the new mixer.
+	 *
+	 * @param {Mixer} mixer - The mixer which will used to play the sound.
+	 */
+	this.play = function(mixer) {
+		this.mixer = mixer;
+		this.playing = true;
+	};
+
+	/**
+	 * Pauses or resumes playback.
+	 *
+	 * This method has no effect if the sound is currently stopped.
+	 *
+	 * @param {Boolean} paused - true to pause, false to resume.
+	 */
+	this.pause = function(paused) {
+		this.playing = !paused;
+	};
+
+	/**
+	 * Stops playing the sound.
+	 *
+	 * Seek position is reset to 0.0s so that the next play() will start at the
+	 * beginning.
+	 */
+	this.stop = function() {
+		this.playing = false;
+		this.position = 0.0;
+		this.mixer = null;
+	};
+}
+exports.Sound = Sound;
```

Comment:
Why not just `export.Sound = function()`? Then you can directly see that the function is exporter. It is only a spec.